src/main/java/no/fint/kulturtanken/service/KulturtankenService.java

```diff
@@ -30,22 +34,37 @@ public class KulturtankenService {
     private final NsrRepository nsrRepository;
     private final FintRepository fintRepository;
     private final KulturtankenProperties kulturtankenProperties;
-
+    private final Clock clock;
+
+    /**
+     * Retrieves the school owner for the given organization ID.
+     * Populates the school owner with schools, levels, and subjects.
+     *
+     * @param orgId the organization ID
+     * @return the {@link Skoleeier} for the given orgId, or null if not found
+     */
     @Cacheable(value = "schoolOwner", unless = "#result == null")
     public Skoleeier getSchoolOwner(String orgId) {
+        // Fetch school owner from NSR
         Skoleeier schoolOwner = nsrRepository.getSchoolOwnerById(orgId);
 
         if (schoolOwner == null) return null;
 
+
+        // If School already exist in kulturtankenProperties, fetch schools from NSR
```

Comment:
Mener vi bør refaktorere koden til å fortelle oss hva som skjer istdenfor å bruke kommentarer. For eksempel kunne vi flyttet denne logikken:
```java
if (kulturtankenProperties.getOrganisations().get(orgId).getSource().equals("nsr")) {
}
```

til en egen metode slik:
```java
private boolean schoolAlreadyExistsInKulturtankenProperties(args...) {
    return kulturtankenProperties.getOrganisations()
        .get(orgId)
        .getSource()
        .equals("nsr");
}
```